### ISSUE TITLE
TestPlatforms.AnyUnix includes Browser

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/runtime#44226]
https://github.com/dotnet/arcade/blob/b5b61c1f0b76c0789a961039413c2cef98cef234/src/Microsoft.DotNet.XUnitExtensions/src/TestPlatforms.cs#L22

We have a lot of tests using `TestPlatforms.AnyUnix`. What should we do about this?

@akoeplinger @ericstj @stephentoub @safern

**Comment by stephentoub** [comment]
Is this why my `[PlatformSpecific(~TestPlatforms.Browser)]` didn't actually take effect?

**Comment by ViktorHofer** [comment]
I'm not sure if that's related, just something that I wanted to have logged to discuss.

> Or if that's not the issue, what is the issue? 😄

Don't know yet. Will let @safern take a closer look.

**Comment by akoeplinger** [comment]
We've been using `[PlatformSpecific(TestPlatforms.AnyUnix & ~TestPlatforms.Browser)]` in other places to disable a test on Browser when using PlatformSpecific.

I agree that ideally AnyUnix shouldn't include Browser, we've discussed that elsewhere in the past.

**Comment by ViktorHofer** [comment]
so here's your answer @stephentoub :)

**Comment by stephentoub** [comment]
> so here's your answer

It is?

**Comment by ViktorHofer** [comment]
> We've been using [PlatformSpecific(TestPlatforms.AnyUnix & ~TestPlatforms.Browser)] in other places to disable a test on Browser when using PlatformSpecific.

I interpret this as Alex and colleagues know about Browser not being excluded unless `TestPlatforms.AnyUnix & ~TestPlatforms.Browser` is specified when using the `PlatformSpecificAttribute`.

**Comment by stephentoub** [comment]
I was doing `[PlatformSpecific(~TestPlatforms.Browser)]`, which would include Windows.  `[PlatformSpecific(TestPlatforms.AnyUnix & ~TestPlatforms.Browser)]` is just saying also don't run on Windows.

**Comment by akoeplinger** [comment]
Ah yeah that should still work of course, otherwise this would be broken already: https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Console/tests/Color.cs#L15 😄

**Comment by safern** [comment]
@akoeplinger -- it seems like it was intentionally added: https://github.com/dotnet/arcade/blob/b5b61c1f0b76c0789a961039413c2cef98cef234/src/Microsoft.DotNet.XUnitExtensions/src/TestPlatforms.cs#L22

https://github.com/dotnet/arcade/commit/e91c8d04275eab1cc57ee8f8a7db530c99429d2d

but it seems like an oversight.

**Comment by ViktorHofer** [comment]
Earlier in net5 Browser derived from Unix which is why this made sense.

**Comment by safern** [comment]
> Earlier in net5 Browser derived from Unix which is why this made sense.

Right, which was a bad assumption to begin with. I might have time to tackle this one... assigning myself and in the case I can't I'll ping the issue.

**Comment by ViktorHofer** [comment]
Thanks 👍
